src/implementation/MassaWalletProviders.ts

```diff
@@ -0,0 +1,106 @@
+import { uid } from 'uid';
+import { ICustomEventMessageResponse } from '../interfaces/ICustomEventMessageResponse';
+import { ICustomEventMessageRequest } from '../interfaces/ICustomEventMessageRequest';
+import { AvailableCommands } from './Commands';
+import { IRegisterEvent } from '../interfaces';
+
+const MASSA_WINDOW_OBJECT_PRAEFIX = 'massaWalletProvider';
+
+type CallbackFunctionVariadicAnyReturn = (
+  error: Error | null,
+  result: any,
+) => any;
+
+// =========================================================
+
+export class MassaWalletProviders {
+  private static instance: MassaWalletProviders;
+  private registeredProviders: { [key: string]: string } = {};
+  private pendingRequests: Map<string, CallbackFunctionVariadicAnyReturn>;
+
+  public static init(): MassaWalletProviders {
+    if (!MassaWalletProviders.instance) {
+      MassaWalletProviders.instance = new MassaWalletProviders();
+    }
+
+    return MassaWalletProviders.instance;
+  }
```

Comment:
just wondering, is it to implement singleton design pattern?

Reply:
It is easier that way esp. since we are exporting the method  `providers(): Provider[]` which automatically inits the client if not already created. I treat it as a convenience pattern.